### PR TITLE
fix(tod.concatenate): Convert index map to unicode.

### PR DESCRIPTION
**Pull request by tristpinsm**

### CODE REVIEW

**Comment by jrs65** [comment]
That would obviously simplify this PR as you could remove two of the chunks.

**Comment by tristpinsm** [comment]
> @tristpinsm do we really need to deal with the case where the concatenation axis has a string type? I struggle to see when a time like axis that is fine for concatenation would have a string type.

Yeah, I dunno. If you did pass it a dataset that fits that description with `convert_dataset_strings=True` wouldn't you expect it to do the conversion? I also can't think of an example where that happens but I don't see any reason why it's impossible.

In any case I'm fine with removing that part, just let me know.

**Review by jrs65 — APPROVED** [review]
This looks good to go for me. I think in the end it makes sense to keep the general behaviour for concatenating along string-types axes, there isn't anything which restricts it in the current implementation and it could be nice in the future.

File: caput/tod.py [hunk 1]
```diff
@@ -350,13 +350,21 @@ def dataset_filter(d):
     for axis, index_map in first_data.index_map.items():
         if axis in concatenation_axes:
             # Initialize the dataset.
-            dtype = index_map.dtype
+            if convert_dataset_strings and memh5.has_bytestring(index_map.dtype):
```
[the remark this answers is not in the document]
Comment (jrs65) [reply caput/tod.py:353]: I was going to say that at least leaving the clause in would save it some work on non- string types, but given that's not true as `.has_bytestring` does a very similar recursive crawl through the dtype structure, so you don't actually save anything. So probably remove it to simplify the implementation.

**Comment by jrs65** [comment]
I think it might rely on the axis being sanely sortable.

**Pull request merged**